File: holi/api_ingles.php
<?php
header("Access-Control-Allow-Origin: *");
header("Access-Control-Allow-Methods: POST");
header("Access-Control-Allow-Headers: Content-Type");
header("Content-Type: application/json");

$api_key = "4b3b890b3d2e4c06a892674f5660f7a6";
$base_url = "https://api.aimlapi.com/v1/chat/completions";
$model = "meta-llama/Llama-Vision-Free";

if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    if (!isset($_FILES['image'])) {
        http_response_code(400);
        echo json_encode(["error" => "No image file provided"]);
        exit;
    }

    $image_tmp = $_FILES['image']['tmp_name'];
    $image_data = file_get_contents($image_tmp);
    $image_base64 = base64_encode($image_data);

    $payload = [
        "model" => $model,
        "messages" => [
            [
                "role" => "user",
                "content" => [
                    [
                        "type" => "text",
                        "text" => "What is the main object or color in this image? Respond ONLY with the object name or color in English. Do not describe or explain. If no object or color is visible, respond with \"None\"."
                    ],
                    [
                        "type" => "image_url",
                        "image_url" => [
                            "url" => "data:image/jpeg;base64," . $image_base64
                        ]
                    ]
                ]
            ]
        ],
        "max_tokens" => 20,
        "temperature" => 0.0
    ];

    $headers = [
        "Authorization: Bearer $api_key",
        "Content-Type: application/json"
    ];

    $ch = curl_init($base_url);
    curl_setopt($ch, CURLOPT_POST, true);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($payload));

    $response = curl_exec($ch);
    $http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    if ($http_code === 200 || $http_code === 201) {
        $result = json_decode($response, true);
        $raw_response = trim($result["choices"][0]["message"]["content"]);

        // Limpia prefijos como "**Answer:**", "Answer:", etc.
        $cleaned = preg_replace('/^\*{0,2}Answer:\*{0,2}\s*/i', '', $raw_response);
        $cleaned = strip_tags($cleaned); // por si acaso contiene HTML

        echo json_encode(["object" => $cleaned]);
    } else {
        http_response_code(500);
        echo json_encode(["error" => "Object detection failed", "details" => $response]);
    }
} else {
    http_response_code(405);
    echo json_encode(["error" => "Invalid request method"]);
}
?>

File: holi/api_ingles.py
import requests
import base64
import pyttsx3

# Datos de la API
api_key = "4b3b890b3d2e4c06a892674f5660f7a6"
base_url = "https://api.aimlapi.com/v1/chat/completions"
model = "meta-llama/Llama-Vision-Free"

# Convierte imagen a base64
def encode_image_to_base64(image_path):
    with open(image_path, "rb") as image_file:
        return base64.b64encode(image_file.read()).decode("utf-8")

# Texto a voz
def speak(text):
    engine = pyttsx3.init()
    engine.say(text)
    engine.runAndWait()

# Enviar imagen a la API y obtener la respuesta
def detect_object_in_image(image_path):
    image_base64 = encode_image_to_base64(image_path)

    headers = {
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json"
    }

    payload = {
        "model": model,
        "messages": [
            {
                "role": "user",
                "content": [
                    {
                        "type": "text",
                        "text": "What object do you see in this image? Answer with only the object name in English."
                    },
                    {
                        "type": "image_url",
                        "image_url": {
                            "url": f"data:image/jpeg;base64,{image_base64}"
                        }
                    }
                ]
            }
        ],
        "max_tokens": 20
    }

    response = requests.post(base_url, headers=headers, json=payload)

    if response.status_code in [200, 201]:  # <-- Cambiado aquí
        result = response.json()
        object_name = result["choices"][0]["message"]["content"].strip()
        return object_name
    else:
        print("Error:", response.status_code, response.text)
        return None

# Programa principal
def main():
    image_path = "imagen.jpg"  # Cambia por la ruta de tu imagen
    print("Analizando imagen...")
    object_name = detect_object_in_image(image_path)

    if object_name:
        print(object_name)
        speak(object_name)

    else:
        speak("There was an error detecting the object.")

if __name__ == "__main__":
    main()
